R/mainTests.r

```diff
@@ -1153,6 +1160,454 @@ country_code <- function(data) {
+      distinct() %>%
+      add_count(name, name = "name_n") %>%
+      add_count(code, name = "code_n")
+
+    multi_count_code <- lookup_creator %>%
+      filter(code_n > 1) %>%
+      mutate(combo = paste0(code, " - ", code_n, " different names")) %>%
+      select(combo) %>%
+      distinct() %>%
+      pull()
+
+    if (length(multi_count_code) == 0) {
+      output <- list(
+        "message" = "every geography code has one assigned geography.",
+        "result" = "PASS"
+      )
+    } else {
+      if (length(multi_count_code) == 1) {
+        output <- list(
+          "message" = paste0("The following geography code has multiple assigned geographies: '", paste0(multi_count_code), "'. 
+                             <br> - Each geography code should have only one assigned geography."),
+          "result" = "FAIL"
+        )
+      } else {
+        if (length(multi_count_code) > 1) {
+          output <- list(
+            "message" = paste0("The following geography codes have multiple assigned geographies: '", paste0(multi_count_code, collapse = "', '"), "'.
+                             <br> - Each geography code should have only one assigned geography."),
+            "result" = "FAIL"
+          )
+        }
+      }
+    }
+  }
+  return(output)
+}
+
+# na_geography -------------------------------------
+# checking if location has code of ":", then name is "not available"
+
+geography_name_codes <- geography_matrix[, 2:3] %>% as.character()
+
+geography_name_codes <- geography_name_codes[!is.na(geography_name_codes)]
+
+
+na_geography <- function(data) {
+  geog_data <- data %>%
+    select(any_of(c(
+      "geographic_level", geography_name_codes
+    ))) %>%
+    distinct() %>%
+    mutate(ID = 1:n())
+
+  names <- geog_data %>%
+    select(ID, geographic_level, contains("name")) %>%
+    mutate_if(is.numeric, as.character) %>%
+    melt(id.vars = c("ID", "geographic_level"), na.rm = TRUE) %>%
+    select(ID, geographic_level, name = value)
+
+  codes <- geog_data %>%
+    select(ID, geographic_level, !contains("name")) %>%
+    mutate_if(is.numeric, as.character) %>%
+    melt(id.vars = c("ID", "geographic_level"), na.rm = TRUE) %>%
+    select(ID, geographic_level, code = value)
+
+  lookup <- names %>%
+    full_join(codes, by = c("ID", "geographic_level")) %>%
+    select(-ID, -geographic_level) %>%
+    distinct()
+
+  check_nas <- lookup %>%
+    subset(code == ":" & !(name == "Not available"))
+
+
+  if (nrow(lookup) == 0) {
+    output <- list(
+      "message" = "No geographies have ':' as a code.",
+      "result" = "IGNORE"
+    )
+  } else {
+    if (nrow(check_nas) == 0) {
+      output <- list(
+        "message" = paste0("All location with a code of ':' have the corresponding name 'Not available'."),
+        "result" = "PASS"
+      )
+    } else {
+      output <- list(
+        "message" = paste0("A location has a code of ':', but does not have the corresponding name 'Not available'."),
```

Comment:
Change to 'At least one location has a code of' to account for single and plural in terms of affected rows?